src/ln/channelmonitor.rs

```diff
@@ -1377,6 +1377,23 @@ impl ChannelMonitor {
 		(Vec::new(), Vec::new())
 	}
 
+	/// Generate a spendable output event when closing transaction get registered onchain, may send
+	/// back two descriptors, so even a false positive one for remote output to avoid watchtower mode
+	/// learning channel balance.
+	fn check_spend_closing_transaction(&self, tx: &Transaction, _height: u32) -> Vec<SpendableOutputDescriptor> {
+		if tx.input[0].sequence == 0xFFFFFFFF && tx.input[0].witness.last().unwrap().len() == 71 {
```

Comment:
I think it's great to relieve users from any chain watching of LN kind of txs ? If not seems to me they have to implement some LN stuff to know it and it lessens keys interface out-of-box value. The other kind of tx which can spend the funding outpoint is commitment tx and the nSequence of their first input is set to an obfuscated commitment number so can't match I would say, but yes I can add heuristic rules by security.

Reply:
I mean they should already have logic for watching an output to a script they generated, the only reason to tell them about outputs is when there is other data mixed in.

Reply:
Aha yes make sense well if so I believe we can get ride of StaticOutputDescriptor (and its tests) ?

Reply:
In fact, checking against bolt, matching is sure (closing tx is the only spending_tx from funding with nSequence set to 0xFFFFFFFF). Don't know if you see my IRC comment but I think my point on offline wallet was irrelevant, detecting this case of output it's still a job handable by ChainWatchInterface if it got destination_script (and if user-provided one want to use a different format descriptor than our, we shouldn't force it to handle our crafted StaticOutputDescriptor). So yes I think that StaticOutputDescriptor can go, the only to reason to keep it could be a performance one (maybe easier to us to filter it, but you know better than me).
For privacy, a rogue watchtower still have the funding_txo and can spot when it spent no ?

Reply:
So on the idea you're ok to keep StaticOutputDescriptor ?

So implemented with closing_signed as of 6fc8391, should avoid to false-positives due to non-definitive closing_signed message exchanged.